doc/OpenMPDevGuide.md

```diff
@@ -0,0 +1,177 @@
+# OpenMP Dev Guide
+This document describes the OpenMP integration from a technical point of view, describing the following:
+- An introduction to the OpenMP runtime library functions and how to patch them
+- The approach used and the modifications to the TAFFO passes
+- The possible next steps to enhance the current integration
+
+Check out [the OpenMP user guide](./OpenMPUserGuide.md) to learn how to use TAFFO with OpenMP.
+
+## Patching OpenMP Runtime
+The OpenMP implementation of the LLVM Framework inserts various indirect functions to manage parallelism;
+moreover, the functions are not contained in the source code that is being compiled.
```

Comment:
```suggestion
these functions are intrinsic, which are not defined in the LLVM-IR module that is being compiled.
```
`source code` is a bit too generic.
`contained` is also not clear: did you refer to definition, call, or declaration? I suppose definition. Correct me if I'm wrong.

Reply:
`contained` refers to the definition indeed, as these functions are just declared in the LLVM IR that is being compiled. I have some doubts regarding the `intrinsic` word. I see that its definition matches in a general context, but the [LLVM LangRef](https://llvm.org/docs/LangRef.html#intrinsic-functions) states that _Intrinsic function names must all start with an “llvm.” prefix_, while the functions we are referring to (such as `__kmpc_fork_call`) do not start with such prefix. Is intrinsic OK or would you suggest something else?

Reply:
In particular, rather than an intrinsic function, we should specify that the `__kmpc_*` functions are added from `clang` in the IR using the -fopenmp command line argument. 
Moreover, the code must be linked to the OpenMP runtime library (that contains the definition of these functions) to run.

Reply:
LLVM intrinsics start with `llvm.`

these can be considered OpenMP instrinsics.

Reply:
Thanks for the clarification, I've updated the vague sentence following your initial suggestion.